MediaBrowser.Common/Net/NetworkExtensions.cs

```diff
@@ -0,0 +1,274 @@
+#pragma warning disable CA1062 // Validate arguments of public methods
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Net;
+using System.Runtime.CompilerServices;
+using System.Text;
+using NetCollection = System.Collections.ObjectModel.Collection<MediaBrowser.Common.Net.IPObject>;
+
+namespace MediaBrowser.Common.Net
+{
+    /// <summary>
+    /// Defines the <see cref="NetworkExtensions" />.
+    /// </summary>
+    public static class NetworkExtensions
+    {
+        /// <summary>
+        /// Add an address to the collection.
+        /// </summary>
+        /// <param name="source">The <see cref="NetCollection"/>.</param>
+        /// <param name="ip">Item to add.</param>
+        public static void AddItem(this NetCollection source, IPAddress ip)
+        {
+            if (!source.ContainsAddress(ip))
+            {
+                source.Add(new IPNetAddress(ip, 32));
+            }
+        }
+
+        /// <summary>
+        /// Adds a network to the collection.
+        /// </summary>
+        /// <param name="source">The <see cref="NetCollection"/>.</param>
+        /// <param name="item">Item to add.</param>
+        public static void AddItem(this NetCollection source, IPObject item)
+        {
+            if (!source.ContainsAddress(item))
+            {
+                source.Add(item);
+            }
+        }
+
+        /// <summary>
+        /// Converts this object to a string.
+        /// </summary>
+        /// <param name="source">The <see cref="NetCollection"/>.</param>
+        /// <returns>Returns a string representation of this object.</returns>
+        public static string AsString(this NetCollection source)
+        {
+            var sb = new StringBuilder();
```

Comment:
Dangling `StringBuilder`

Reply:
Feel free to correct me should I be Wrong here, but isn't this the same SB that i noticed in `Networking 2`?

Reply:
I believe Networking 2 has all of Networking 1 changes + more

Reply:
Yes it has - i've tried to add comments to all that are duplicated. - it's the only way to get the code to compile - whilst keeping the PR small. 

Did think PR 1 would be accepted & merged first, thus reducing the size of PR2.